consensus/validation.go

```diff
@@ -63,19 +63,22 @@ func validateMinerPayouts(s State, b types.Block) error {
 			return errors.New("block must have exactly one miner payout")
 		}
 	}
-
-	var sum types.Currency
-	for _, mp := range b.MinerPayouts {
-		if mp.Value.IsZero() {
-			return errors.New("miner payout has zero value")
+	if s.childHeight() < s.Network.HardforkV2.FinalCutHeight {
+		var sum types.Currency
+		for _, mp := range b.MinerPayouts {
+			if mp.Value.IsZero() {
+				return errors.New("miner payout has zero value")
+			}
+			sum, overflow = sum.AddWithOverflow(mp.Value)
+			if overflow {
+				return errors.New("miner payouts overflow")
+			}
 		}
-		sum, overflow = sum.AddWithOverflow(mp.Value)
-		if overflow {
-			return errors.New("miner payouts overflow")
+		if sum != expectedSum {
+			return fmt.Errorf("miner payout sum (%v) does not match block reward + fees (%v)", sum, expectedSum)
 		}
-	}
-	if sum != expectedSum {
-		return fmt.Errorf("miner payout sum (%v) does not match block reward + fees (%v)", sum, expectedSum)
+	} else if !b.MinerPayouts[0].Value.IsZero() {
+		return errors.New("miner payout must be zero after finalcut hardfork")
```

Comment:
if we want to avoid making any extra work for miners/pools, this change probably isn't tenable :/

Reply:
decided to relax it to "the payout value _may_ be omitted." Hopefully in the long run we'll observe that everyone is omitting the value, and then we can have an uncontroversial fork that enforces the omission.

for context, the main reason I want to get rid of this (besides it being redundant) is that, even in v2 blocks, the miner payout values use the v1 currency encoding. In fact, they are the _only_ place in v2 that uses the old encoding, so it annoys me that we have to maintain this one special case. At this point, we can't change the block encoding itself, but we _can_ require the value to be omitted entirely, which accomplishes the same goal of enabling us to completely strip out the v1 currency code.